docs/source/chatops/msteams_is_a_diva.rst

```diff
@@ -0,0 +1,234 @@
+Using Microsoft Teams (with BotFramework)
+=========================================
+
+Configurating st2chatops with Microsoft Teams is **much** more involved than configuring
+other chat platforms.
+
+.. note:: Caveats
+
+    * There are multiple hubot adapters for Microsoft Teams and/or BotFramework. Only
+      two of them (that we found) are production quality, and only the
+      BotFramework-Hubot project is the officially supported option.
+    * Microsoft BotFramework's "push model" requires bots to have an open port open to
+      the internet. This is a limitation of Teams/BotFramework.
+      (`source <https://github.com/Microsoft/BotFramework-Hubot#common-differences-in-hubot-running-in-slack-hipchat-other-chat-platforms-and-ms-teams>`_)
+    * The BotFramework bot does not receive all messages in a channel. Users MUST
+      explicitly at-mention the bot user to send any messages to the bot.
+    * Users can alternatively direct message the bot user itself (without
+      at-mentioning it) in the bot user's "room". This isn't really considered ChatOps
+      though. And it is impossible to disable this "feature" from within Teams itself.
+      However, all StackStorm alias executions are logged as normal, although StackStorm's
+      RBAC does not work in st2chatops.
+    * The `Orky <https://github.com/OfficeDev/Orky>`_ project is meant to run within Azure
+      and send events via a websocket. We have investigated this method but we were unable
+      to get it working in development.
+
+Setup BotFramework
+~~~~~~~~~~~~~~~~~~
+
+1. Note the internet accessible hostname and port for your st2chatops (hubot) server.
+2. Register a bot within `BotFramework <https://dev.botframework.com/bots/new>`_.
+3. Note your Microsoft App ID and password, you will need these later.
+4. Configure the "Messaging endpoint" option to your st2chatops server, and append
```

Comment:
Some screenshots wouldn't hurt to make it even more "dummy" proof.

Reply:
Good idea, but I'll handle that later. Right now, even though it's officially supported, I don't consider the adapter 100% GA, more like beta quality. Once it gets to GA status I'll circle back to this.

Although hopefully we have Orky working by then.